server/src/main/java/io/crate/metadata/IndexReference.java

```diff
@@ -82,7 +85,10 @@ public IndexReference build() {
 
     @Nullable
     private final String analyzer;
+    @Nonnull
     private final List<Reference> columns;
+    @Nonnull
+    private final List<String> columnNames;
```

Comment:
As discussed in chat,  please don't duplicate the information.  `columns ` is a superset of columnNames